### PR TITLE
AWS API Module

### PR DESCRIPTION
**Pull request by tallieber**
## Status
- [x] Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/32958
relates: https://github.com/demisto/etc/issues/34707

## Description
Generic infra for all AWS integrations. Add timeout and retries parameters to integration configuration

## Minimum version of Cortex XSOAR
- [x] 5.5.0

## Does it break backward compatibility?
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/ApiModules/Scripts/AWSApiModule/AWSApiModule.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+            read_timeout=read_timeout,
+            retries=dict(
+                max_attempts=int(retries)
+            ),
+            proxies=proxies
+        )
+
+    """HELPER FUNCTIONS"""
+
+    def aws_session(self, service, region=None, role_arn=None, role_session_name=None, role_session_duration=None,
```
Comment (bakatzir) [inline Packs/ApiModules/Scripts/AWSApiModule/AWSApiModule.py:42]: add type annotations.
also - all of theses args override the class props if given? let's add docstrings and explain that.

**Review by bakatzir — CHANGES_REQUESTED** [review]
I haven't seen to much of error handling, did you check that?
also, are we adding UT to check that upon giving the wrongs creds / non existent region / etc we are producing a meaningful error?

File: Packs/ApiModules/Scripts/AWSApiModule/AWSApiModule.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+type: python
+subtype: python3
+tags:
+- infra
+- server
+comment: AWS Client class, provides generic Infrastructure to all AWS integrations
+scripttarget: 0
+dependson: {}
+timeout: 0s
+dockerimage: demisto/boto3:2.0.0.19445
```
Comment (amshamah419) [inline Packs/ApiModules/Scripts/AWSApiModule/AWSApiModule.yml:15]: Isn't this a python2 image?

Comment (bakatzir) [reply Packs/ApiModules/Scripts/AWSApiModule/AWSApiModule.yml:15]: yes, as 4/8 integrations are in py2.

File: Packs/ApiModules/Scripts/AWSApiModule/AWSApiModule.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+                else:
+                    client = boto3.client(
+                        service_name=service,
+                        region_name=self.aws_default_region,
+                        aws_access_key_id=sts_response['Credentials']['AccessKeyId'],
+                        aws_secret_access_key=sts_response['Credentials']['SecretAccessKey'],
+                        aws_session_token=sts_response['Credentials']['SessionToken'],
+                        verify=self.verify_certificate,
+                        config=self.config
+                    )
```
Comment (bakatzir) [inline Packs/ApiModules/Scripts/AWSApiModule/AWSApiModule.py:95]:
```suggestion
                    client = boto3.client(
                        service_name=service,
                        region_name=region if region else self.aws_default_region,
                        aws_access_key_id=sts_response['Credentials']['AccessKeyId'],
                        aws_secret_access_key=sts_response['Credentials']['SecretAccessKey'],
                        aws_session_token=sts_response['Credentials']['SessionToken'],
                        verify=self.verify_certificate,
                        config=self.config
                    )
```

File: Packs/ApiModules/Scripts/AWSApiModule/AWSApiModule.py [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+                )
+            else:
+                client = boto3.client(
+                    service_name=service,
+                    region_name=self.aws_default_region,
+                    aws_access_key_id=self.aws_access_key_id,
+                    aws_secret_access_key=self.aws_secret_access_key,
+                    verify=self.verify_certificate,
+                    config=self.config
+                )
```
Comment (bakatzir) [inline Packs/ApiModules/Scripts/AWSApiModule/AWSApiModule.py:136]:
```suggestion
                client = boto3.client(
                    service_name=service,
                    region_name=region if region else self.aws_default_region,
                    aws_access_key_id=self.aws_access_key_id,
                    aws_secret_access_key=self.aws_secret_access_key,
                    verify=self.verify_certificate,
                    config=self.config
                )
            else:
                client = boto3.client(
                    service_name=service,
                    region_name=self.aws_default_region,
                    aws_access_key_id=self.aws_access_key_id,
                    aws_secret_access_key=self.aws_secret_access_key,
                    verify=self.verify_certificate,
                    config=self.config
                )
```

**Review by bakatzir — APPROVED** [review]

**Pull request merged**